pyclean/modern.py

```diff
@@ -252,6 +252,31 @@ def remove_freeform_targets(glob_patterns, yes, directory):
         delete_filesystem_objects(directory, path_glob, prompt=not yes)
 
 
+def delete_debris_recursive(directory, patterns):
+    """
+    Recursively delete debris matching any of the given patterns.
+    
+    This function walks the directory tree once and applies all patterns
+    at each level, avoiding redundant directory scans.
+    """
+    # Apply all patterns at the current directory level
+    for pattern in patterns:
+        delete_filesystem_objects(directory, pattern, prompt=False, recursive=False)
+    
+    # Recursively descend into subdirectories (except ignored ones)
```

Comment:
Remove this comment. This is a trivial correlation for anyone with experience in recursive algorithms.